test/css/component_selector_use_test.rb

```diff
@@ -88,8 +88,12 @@ class ComponentSelectorUseTest < System::TestCase
 
           const nodeMatchesAtLeastOneComponentSelector = (node) => {
             for (const componentSelector of componentSelectors) {
-              if (node.matches(componentSelector) || node.querySelectorAll(`:scope > ${componentSelector}`).length > 0) {
-                return true;
+              try {
+                if (node.matches(componentSelector) || node.querySelectorAll(`:scope > ${componentSelector}`).length > 0) {
+                  return true;
+                }
+              } catch (e) {
+                return false;
```

Comment:
This is needed because `:open` is not currently a valid selector. The CSS parser in the browser will safely ignore this rule, but `matches`/`qSA` will throw. Given these are CSS selectors only, we just want to return false instead of throw.